src/sensors.js

```diff
@@ -1,68 +1,49 @@
-import { NativeModules, DeviceEventEmitter } from "react-native";
-import * as Rx from "rxjs/Rx";
-const { Gyroscope: GyroNative, Accelerometer: AccNative } = NativeModules;
-
-const handle = {
-  Accelerometer: AccNative,
-  Gyroscope: GyroNative
-};
-
-const RNSensors = {
-  start: function(type, updateInterval) {
-    const api = handle[type];
-    api.setUpdateInterval(updateInterval);
-    // A promise is returned in Android, since it can fail with an exception
-    return api.startUpdates() || Promise.resolve();
-  },
-
-  stop: function(type) {
-    const api = handle[type];
-    api.stopUpdates();
-  }
-};
+import { DeviceEventEmitter } from 'react-native';
+import * as Rx from 'rxjs/Rx';
+import RNSensors from './rnsensors'
 
 function createSensorMonitorCreator(sensorType) {
   function Creator(options = {}) {
-    const {
-      updateInterval = 100 // time in ms
-    } =
-      options || {};
-
-    let observer;
-
-    // Instanciate observable
-    const observable = Rx.Observable.create(function(obs) {
-      observer = obs;
-      // Start the sensor manager
-      RNSensors.start(sensorType, updateInterval).then(
-        () => {
-          DeviceEventEmitter.addListener(sensorType, function(data) {
-            observer.next(data);
+    return new Promise((resolve, reject) => {
```

Comment:
We could also use promise chaining here, right? 


```js
function Creator(options = {}) {
	return RNSensors.isAvailable(sensorType).then(() => {
		// same as before
		return observable;
	})
}
```

That way we don't need to wrap it inside a new Promise and can leave out the catch 👍

Reply:
Yes, right, we coudl use promise chaining here. Done!